### PR TITLE
Update weekly checkout order summary to match the design

### PR DESCRIPTION
**Pull request by lucymonie**
## Why are you doing this?

This adds an order summary for the new weekly checkout that matches the one in paper and digital pack. The work is detailed in this [**Trello Card**](https://trello.com/c/BGfcbd5L/2376-guardian-weekly-order-summary)

### Mobile closed
Screen Shot 2019-05-20 at 15 24 53

### Mobile open
Screen Shot 2019-05-20 at 15 25 01

### Desktop
Screen Shot 2019-05-20 at 15 25 16

### CODE REVIEW

File: support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx [hunk 1]
```diff
@@ -98,10 +99,11 @@ const days = getDisplayDays();
 
 // ----- Component ----- //
 type Plans = {
-  [WeeklyBillingPeriod]: {
+  [WeeklyBillingPeriod | typeof Monthly]: {
```
Comment (rupertbates) [inline support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: Why does this need to accept 'Monthly'?

Comment (lucymonie) [reply support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: Flow was chucking a hissy fit about it - I'm still not clear why it needs it.

File: support-frontend/assets/components/subscriptionCheckouts/summary.jsx [hunk 2]
```diff
@@ -215,13 +217,17 @@ export default class Summary extends Component<PropTypes, StateTypes> {
   }
 
   render() {
+    const url = new URL(window.location).pathname;
```
Comment (rupertbates) [inline support-frontend/assets/components/subscriptionCheckouts/summary.jsx]: I think a better way to do this is to pass the `product` from the state into this component (you'll need to rebase onto master to get this field).

Comment (lucymonie) [reply support-frontend/assets/components/subscriptionCheckouts/summary.jsx]: Sure, shall do

File: support-frontend/assets/helpers/productPrice/weeklyProductPrice.js [hunk 3]
```diff
@@ -5,14 +5,22 @@ import { currencies, detect } from 'helpers/internationalisation/currency';
...
 
 
-const getPrice = (countryGroupId: CountryGroupId, period: WeeklyBillingPeriod) => [
+export const getPrice = (countryGroupId: CountryGroupId, period: WeeklyBillingPeriod) => [
   currencies[detect(countryGroupId)].extendedGlyph,
   getWeeklyProductPrice(countryGroupId, period),
 ].join('');
 
-const getPromotionPrice = (countryGroupId: CountryGroupId, period: WeeklyBillingPeriod, promoCode: string) => [
+export const getCurrencyAndPrice = (countryGroupId: CountryGroupId, period: WeeklyBillingPeriod): Price => (
```
Comment (rupertbates) [inline support-frontend/assets/helpers/productPrice/weeklyProductPrice.js]: There's going to have to be some changes to this code because it's using hard coded values in subscriptions.js at the moment and it should be using the prices from the `productPrices` object in the same way as the other two checkouts, but that can be a separate PR.

File: support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx [hunk 4]
```diff
@@ -146,12 +151,16 @@ function WeeklyCheckoutForm(props: PropTypes) {
           }
           title="Guardian Weekly"
           description=""
-          productPrice={{ price: 99.89, currency: 'GBP' }}
+          productPrice={summaryPrice}
           promotion={undefined}
```
Comment (rupertbates) [inline support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: We're going to need to pass in the promotion here

File: support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx [hunk 5]
```diff
@@ -146,12 +151,16 @@ function WeeklyCheckoutForm(props: PropTypes) {
...
           dataList={[
+            {
+              title: 'Delivery method',
+              value: 'Home delivery',
+            },
           ]}
-          billingPeriod="Monthly"
-          changeSubscription="TODO placeholder"
+          billingPeriod={props.billingPeriod}
+          changeSubscription="/subscribe/weekly"
```
Comment (rupertbates) [inline support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: Can use `guardianWeeklySubscriptionLanding` from routes.js here

File: support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx (hunk 1 above)
Comment (rupertbates) [reply support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: Ah it's because the incoming prop is of type `BillingPeriod`, I'd just change this type to `BillingPeriod` to match.

File: support-frontend/assets/helpers/productPrice/weeklyProductPrice.js (hunk 3 above)
Comment (lucymonie) [reply support-frontend/assets/helpers/productPrice/weeklyProductPrice.js]: 👍

File: support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx (hunk 4 above)
Comment (lucymonie) [reply support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: Yes - as discussed, I'll leave this one until after your weekly pricing rejig is complete.

File: support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx (hunk 5 above)
Comment (lucymonie) [reply support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.jsx]: 👍
